lib/features/popup-menu/AppendMenuProvider.js

```diff
@@ -0,0 +1,166 @@
+import { isUndefined } from 'min-dash';
+import { CREATE_OPTIONS } from './util/OptionsUtil';
+
+/**
+ * This module is an append menu provider for the popup menu.
+ */
+export default function AppendMenuProvider(
+    elementFactory,
+    popupMenu,
+    create,
+    autoPlace,
+    rules,
+    mouse
+) {
+  this._elementFactory = elementFactory;
+  this._popupMenu = popupMenu;
+  this._create = create;
+  this._autoPlace = autoPlace;
+  this._rules = rules;
+  this._create = create;
+  this._mouse = mouse;
+
+  this.register();
+}
+
+AppendMenuProvider.$inject = [
+  'elementFactory',
+  'popupMenu',
+  'create',
+  'autoPlace',
+  'rules',
+  'mouse'
+];
+
+/**
+ * Register append menu provider in the popup menu
+ */
+AppendMenuProvider.prototype.register = function() {
+  this._popupMenu.registerProvider('bpmn-append', this);
+};
+
+/**
+ * Get all entries from createOptions for the given element.
+ *
+ * @param {djs.model.Base} element
+ *
+ * @return {Array<Object>} a list of menu entry items
+ */
+AppendMenuProvider.prototype.getEntries = function(element) {
```

Comment:
Let's use the new version `PopupMenuProvider#getPopupMenuEntries`.

```suggestion
AppendMenuProvider.prototype.getPopupMenuEntries = function(element) {
```